core/translate/expr.rs

```diff
@@ -967,6 +967,56 @@ pub fn translate_expr(
 
                             Ok(target_register)
                         }
+                        ScalarFunc::Iif => {
+                            let args = match args {
+                                Some(args) if args.len() == 3 => args,
+                                _ => crate::bail_parse_error!(
+                                    "{} requires exactly 3 arguments",
+                                    srf.to_string()
+                                ),
+                            };
+                            let temp_reg = program.alloc_register();
+                            translate_expr(
+                                program,
+                                referenced_tables,
+                                &args[0],
+                                temp_reg,
+                                precomputed_exprs_to_registers,
+                            )?;
+                            let jump_target_when_false = program.allocate_label();
+                            program.emit_insn_with_label_dependency(
+                                Insn::IfNot {
+                                    reg: temp_reg,
+                                    target_pc: jump_target_when_false,
+                                    null_reg: 1,
```

Comment:
I don't quite understand what the null_reg is about, so this isn't right, but I'm not sure what is.

> If the value in P1 is NULL then take the jump if and only if P3 is non-zero.

Would make me think I'm supposed to allocate a register and load it with a non-zero constant for this, but the equivalent sqlite code just points at `1` which was never touched or initialized or anything.  So, I don't quite get it.

Reply:
I don't precisely know when it's used with a 0 and non-0, but a high level comment here:https://github.com/penberg/limbo/pull/425#discussion_r1876253256